dds/DCPS/XTypes/DynamicDataImpl.cpp

```diff
@@ -4654,12 +4654,50 @@ void serialized_size_wstring_value(const Encoding& encoding, size_t& size, const
 }
 #endif
 
-bool check_rc_from_get(DDS::ReturnCode_t rc, DDS::MemberId id, DDS::TypeKind tk, const char* fn_name)
+const char* log_level_to_string(LogLevel::Value log_level)
```

Comment:
This function and the one below aren't related to DynamicData - could they be moved somewhere else? Or, do they already exist somewhere else in the library?

Reply:
There seems no similar function elsewhere. I moved them to XTypes/Utils.h. We could move them to a common utility file in dds/DCPS but there seems no use for them right now in DCPS.

Reply:
Is it the same as `LogLevel::get_as_string()`?

Reply:
It's very similar. Maybe make them static member functions in `LogLevel`?

Reply:
Given a value `v` of type `LogLevel::Value`, the existing code appears to support `LogLevel(v).get_as_string()`

Reply:
Yes, I realized that as well. I was hesitant to use it this way because the constructor of `LogLevel` does other setups that aren't relevant to the purpose of `to_string`. The downside though is that there is some similarity between `get_as_string` and `to_string`.

@iguessthislldo Fred, can you take a look at these changes to `LogLevel`?

Reply:
We shouldn't change the spelling of the levels like that because they're part of the public API documented in the devguide. Maybe add optional boolean argument to `get_as_string` to capitalize the name? I'm also fine with the non-static method called a static common function.